### PR TITLE
[Feature] Implement the Local Avatar into Composite

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
- Include new Avatar picker into settings page
- Introduce AvatarManager
- Inject AvatarManager into the Setup View and Call View
- Loading avatar into the local participant 
Screen Shot 2022-04-18 at 1 43 33 PM
Screen Shot 2022-04-18 at 1 43 44 PM
Screen Shot 2022-04-18 at 1 43 59 PM
Screen Shot 2022-04-18 at 1 44 11 PM

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Step 1: Load up to sample app
Step 2: Go to settings page and pick an avatar
Step 3: Join a group call
Step 4: Notice Avatar in the set up view / call view

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Views/UIKitDemoViewController.swift [hunk 1]
```diff
@@ -142,6 +142,10 @@ class UIKitDemoViewController: UIViewController {
         let callComposite = CallComposite(withOptions: callCompositeOptions)
 
         callComposite.setTarget(didFail: didFail)
+        let persona = CommunicationUIPersonaData(UIImage(named: envConfigSubject.avatarImageName),
+                                                 renderDisplayName: envConfigSubject.renderedDisplayName.isEmpty ?
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Views/UIKitDemoViewController.swift:146]: Can you create a variable for `envConfigSubject.renderedDisplayName.isEmpty ? nil : envConfigSubject.renderedDisplayName` so there won't be mixing of initializar parameters and conditional operator?

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Views/UIKitDemoViewController.swift]: Done

**Review by palatter — APPROVED** [review]
Looks good and tested.

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Assets.xcassets/Avatars/monkey.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "filename" : "monkey4x-1.png",
+      "idiom" : "universal",
+      "scale" : "2x"
+    },
+    {
+      "filename" : "monkey4x.png",
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Assets.xcassets/Avatars/monkey.imageset/Contents.json:14]: It looks like the <name>4x.png files are duplicated in avatar folder (navigate avatar asset folder in file manager). Do we need them?

**Review by vhuseinova-msft — APPROVED** [review]
A minor comment, otherwise looks good!

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Assets.xcassets/Avatars/monkey.imageset/Contents.json (hunk 2 above)
Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Assets.xcassets/Avatars/monkey.imageset/Contents.json:14]: No but we can resolve this in an upcoming PR. I have a task to update this to svg's

**Pull request merged**